code/framework/encryption/src/main/java/io/cattle/platform/framework/encryption/impl/Aes128Encrypter.java

```diff
@@ -0,0 +1,72 @@
+package io.cattle.platform.framework.encryption.impl;
+
+import io.cattle.platform.framework.encryption.Encrypter;
+import io.cattle.platform.framework.encryption.EncryptionUtils;
+
+import java.security.Key;
+import java.security.SecureRandom;
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.codec.binary.Hex;
+
+public class Aes128Encrypter extends Encrypter {
+    private String encryptionKey;
+
+    private Cipher encrypter;
+    private Cipher decrypter;
+    private static final String NAME = "AES128";
+    private Key key;
+    private SecureRandom rn;
+
+    public Aes128Encrypter() throws Exception {
+        rn = new SecureRandom();
+
+        encrypter = Cipher.getInstance("AES/CBC/PKCS5PADDING", "SunJCE");
+        try {
+            encryptionKey = EncryptionUtils.getKeyFromFile(NAME + "Key");
+        } catch (NullPointerException e) {
```

Comment:
Would renaming the function to getOrCreateKeyFromFile and making the function generate and save a key if it can't get one be a good option as well? basically taking the logic here and moving it into encryption utils.